leakpro/tests/anonymeter_tests/test_confidence.py

```diff
@@ -0,0 +1,251 @@
+"""Tests for confidence module."""
+# This file is part of Anonymeter and is released under BSD 3-Clause Clear License.
+# Copyright (c) 2022 Anonos IP LLC.
+# See https://github.com/statice/anonymeter/blob/main/LICENSE.md for details.
+from typing import Union
+
+import numpy as np
+import pytest
+from pydantic_core._pydantic_core import ValidationError
+
+from leakpro.synthetic_data_attacks.anonymeter.stats.confidence import (
+    EvaluationResults,
+    SuccessRate,
+    assert_x_in_bound,
+    get_confidence_interval,
+    residual_rate,
+    success_rate,
+)
+
+
+def test_assert_x_in_bound() -> None:
+    """Assert results of assert_x_in_bound function on different input."""
+    #Case not correct type input
+    with pytest.raises(TypeError) as e:
+        assert_x_in_bound(x="not_float_or_int")
+    assert e.type == TypeError
+    assert str(e.value) == "'<=' not supported between instances of 'str' and 'float'"
+    #Case ValueError when x not in bounds
+    e_msg = "Parameter `param1` must be > 0.0 and < 10. Got 11 instead."
+    with pytest.raises(ValueError, match=e_msg) as e:
+        assert_x_in_bound(x=11, x_name="param1", high_bound=10)
+    assert e.type == ValueError
+    #Case x in bounds
+    assert_x_in_bound(x=0.5)
+    assert_x_in_bound(x=9, high_bound=10)
+    #Case x equal to low_bound/high_bound and inclusive_flag = True
+    assert_x_in_bound(x=0, inclusive_flag=True)
+    assert_x_in_bound(x=1, inclusive_flag=True)
+    #Case x equal to low_bound/high_bound and inclusive_flag = True
+    e_msg = "Parameter must be > 0.0 and < 1.0. Got 0 instead."
+    with pytest.raises(ValueError, match=e_msg) as e:
+        assert_x_in_bound(x=0, inclusive_flag=False)
+    assert e.type == ValueError
+    e_msg = "Parameter must be > 0.0 and < 1.0. Got 1 instead."
+    with pytest.raises(ValueError, match=e_msg) as e:
+        assert_x_in_bound(x=1, inclusive_flag=False)
+    assert e.type == ValueError
+
+@pytest.mark.parametrize(
+    ("rate", "error", "e_lower_b", "e_upper_b"),
+    [
+        (0.1, 0.2, 0, 0.3),
+        (0.9, 0.15, 0.75, 1),
+        (0.5, 0.1, 0.4, 0.6)
+    ],
+)
+def test_get_confidence_interval(rate: float, error: float, e_lower_b: float, e_upper_b: float) -> None:
+    """Assert results of get_confidence_interval function on different input."""
+    lower_b, upper_b = get_confidence_interval(rate=rate, error=error)
+    assert round(lower_b, 3) == e_lower_b
+    assert round(upper_b, 3) == e_upper_b
+
+def test_SuccessRate_init() -> None:
```

Comment:
we should not use capital letters here. I think its better to use success_rate as in the next test.